From a49a43e3cc88e958f53d04f3a7ac41b3e81304d5 Mon Sep 17 00:00:00 2001
From: Kjetil Matheussen <k.s.matheussen@notam02.no>
Date: Thu, 5 Mar 2020 12:56:36 +0100
Subject: [PATCH 2/3] Sequencer: Fix program sometimes crashing in debug mode
 when finalizing a seqblock or seqtrack. Only seen it happen on windows32 for
 some reason.

---
 common/sequencer.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/common/sequencer.cpp b/common/sequencer.cpp
index ff2616678..2149fa5bb 100644
--- a/common/sequencer.cpp
+++ b/common/sequencer.cpp
@@ -2236,8 +2236,10 @@ void SEQTRACK_delete_seqblock(struct SeqTrack *seqtrack, const struct SeqBlock *
 
 #if !defined(RELEASE)
   //memset((void*)seqblock, 0, sizeof(struct SeqBlock));
+  /*
   if(seqblock->block != NULL) // Can't release sample seqblock memory here since it may still be used in Seqtrack_plugin.cpp.
-    tfree((void*)seqblock); // Only for debugging.
+    tfree((void*)seqblock); // Only for debugging. // Commented out, can't tfree memory that has a finalizer.
+  */
 #endif
 }
 
@@ -3206,7 +3208,7 @@ void SEQUENCER_delete_seqtrack(int pos){
 
 #if !defined(RELEASE)
   //memset(old_seqtrack, 0, sizeof(struct SeqTrack));
-  tfree((void*)old_seqtrack);
+  // tfree((void*)old_seqtrack);  // Commented out, can't tfree memory that has a finalizer.
 #endif
 }
 
-- 
2.25.1

